src/DI/ContainerBuilder.php

```diff
@@ -78,7 +78,22 @@ public function addDefinition($name, ServiceDefinition $definition = NULL)
 	public function removeDefinition($name)
 	{
 		$name = isset($this->aliases[$name]) ? $this->aliases[$name] : $name;
-		unset($this->definitions[$name]);
+
+		if (isset($this->definitions[$name])) {
+			if ($this->classes) {
+				foreach ($this->classes as $class => $tmp) {
+					foreach ($tmp as $mode => $serviceNames) {
+						foreach ($serviceNames as $key => $serviceName) {
```

Comment:
Hmmm, and what about this? `$this->classes[$class][$mode] = array_diff($this->classes[$class][$mode], $name)`.